core/miner/storage_fsm/impl/sealing_impl.cpp

```diff
@@ -409,6 +411,51 @@ namespace fc::mining {
     return outcome::success();
   }
 
+  outcome::result<void> SealingImpl::markForSnapUpgrade(SectorNumber id) {
+    // TODO(a.chernyshov)
+    // https://github.com/filecoin-project/lotus/blob/362c73bfbdb8c6d5f1d110b25ee33faa2b5c8dcc/extern/storage-sealing/upgrade_queue.go#L53-L62
+    // check staging - waitDealsSectors < config.maxWaitDealsSectors
+
+    OUTCOME_TRY(sector_info, getSectorInfo(id));
+
+    if (sector_info->state != SealingState::kProving) {
+      return SealingError::kNotProvingState;
+    }
+    if (sector_info->pieces.size() != 1) {
+      return SealingError::kUpgradeSeveralPieces;
+    }
+    if (sector_info->pieces[0].deal_info.has_value()) {
+      return SealingError::kUpgradeWithDeal;
+    }
+
+    OUTCOME_TRY(head, api_->ChainHead());
+    OUTCOME_TRY(active_sectors,
+                api_->StateMinerActiveSectors(miner_address_, head->key));
+    // Ensure the upgraded sector is active
+    if (find_if(active_sectors.begin(),
+                active_sectors.end(),
+                [id](const auto &sector) { return sector.sector == id; })
+        == active_sectors.end()) {
+      return SealingError::kCannotMarkInactiveSector;
+    }
+
+    OUTCOME_TRY(onchain_info,
+                api_->StateSectorGetInfo(miner_address_, id, head->key));
+    if (onchain_info->expiration - head->epoch() < kDealMinDuration) {
+      logger_->error(
+          "pointless to upgrade sector {}, expiration {} is less than a min "
+          "deal duration away from current epoch. Upgrade expiration before "
+          "marking for upgrade",
+          id,
+          onchain_info->expiration);
+      return SealingError::kSectorExpirationError;
+    }
+
+    FSM_SEND(sector_info, SealingEvent::kSectorStartCCUpdate);
```

Comment:
Need context

Apply event:
```go
state.CCUpdate = true
// Clear filler piece but remember in case of abort
state.CCPieces = state.Pieces
state.Pieces = nil
```

Reply:
done